quantum/process_keycode/process_unicode_common.c

```diff
@@ -153,3 +147,56 @@ void send_unicode_hex_string(const char *str) {
     str += n; // Move to the first ' ' (or '\0') after the current token
   }
 }
+
+bool process_record_unicode_common(uint16_t keycode, keyrecord_t *record) {
+  if (record->event.pressed) {
+    switch (keycode) {
+      case UNICODE_MODE_OSX:
+      set_unicode_input_mode(UC_OSX);
+        #if defined(AUDIO_ENABLE) && defined(UNICODE_SONG_OSX)
+          PLAY_SONG(osx_song);
+        #endif
+        break;
+      case UNICODE_MODE_LNX:
+        set_unicode_input_mode(UC_LNX);
+        #if defined(AUDIO_ENABLE) && defined(UNICODE_SONG_LNX)
+          PLAY_SONG(song_lnx);
+        #endif
+        break;
+      case UNICODE_MODE_BSD:
+        set_unicode_input_mode(UC_BSD);
+        #if defined(AUDIO_ENABLE) && defined(UNICODE_SONG_BSD)
+          PLAY_SONG(song_bsd);
+        #endif
+        break;
+      case UNICODE_MODE_WIN:
+        set_unicode_input_mode(UC_WIN);
+        #if defined(AUDIO_ENABLE) && defined(UNICODE_SONG_WINDOWS)
+          PLAY_SONG(windows_song);
+        #endif
+        break;
+      case UNICODE_MODE_WINC:
+        set_unicode_input_mode(UC_WINC);
+        #if defined(AUDIO_ENABLE) && defined(UNICODE_SONG_WIN_COMPOSE)
+          PLAY_SONG(win_compose_song);
+        #endif
+        break;
+      case UNICODE_MODE_OSX_RALT:
+        set_unicode_input_mode(UC_OSX_RALT);
+        #if defined(AUDIO_ENABLE) && defined(UNICODE_SONG_OSX_RALT)
+          PLAY_SONG(osx_ralt_song);
+        #endif
+        break;
+    }
+  }
+  #ifdef UNICODE_ENABLE
```

Comment:
I think it'd be a good idea to error out at compile time if more than one of these is defined.

```suggestion
  #if defined(UNICODE_ENABLE) + defined(UNICODEMAP_ENABLE) + defined(UCIS_ENABLE) > 1
    #error "Cannot enable more than one of UNICODE, UNICODEMAP, UCIS at the same time"
  #endif
  #ifdef UNICODE_ENABLE
```